Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -41,5 +61,46 @@ ComponentInfo::detectorIndices(const size_t componentIndex) const {
 
 size_t ComponentInfo::size() const { return m_size; }
 
+Eigen::Vector3d ComponentInfo::position(const size_t componentIndex) const {
+  if (isDetectorDomain(componentIndex)) {
+    return m_detectorInfo->position(componentIndex);
+  }
+  const auto rangesIndex =
+      componentIndex - m_assemblySortedDetectorIndices->size();
+  return (*m_positions)[rangesIndex];
+}
+
+Eigen::Quaterniond ComponentInfo::rotation(const size_t componentIndex) const {
+  if (isDetectorDomain(componentIndex)) {
+    return m_detectorInfo->rotation(componentIndex);
+  }
+  const auto rangesIndex =
+      componentIndex - m_assemblySortedDetectorIndices->size();
+  return (*m_rotations)[rangesIndex];
+}
+
+void ComponentInfo::setPosition(const size_t componentIndex,
+                                const Eigen::Vector3d &position) {
+  if (isDetectorDomain(componentIndex)) {
+    m_detectorInfo->setPosition(componentIndex, position);
+  }
+  const auto rangesIndex =
+      componentIndex - m_assemblySortedDetectorIndices->size();
+  m_positions.access()[rangesIndex] = position;
+}
+
+void ComponentInfo::setRotation(const size_t componentIndex,
+                                const Eigen::Quaterniond &rotation) {
+
+  if (isDetectorDomain(componentIndex)) {
+    m_detectorInfo->setRotation(componentIndex, rotation);
+  }
+  const auto rangesIndex =
+      componentIndex - m_assemblySortedDetectorIndices->size();
+  m_rotations.access()[rangesIndex] = rotation; // TODO adjust position too.
```

Comment:
This should update the positions and rotations of all components in the subtree, I think.